balancer-labs/frontend-v2
# Fetch Pools using Balancer API + SDK

# Description

This PR:
- pulls the initial pool list from the new Balancer GraphQL API using the SDK's PoolsBalancerApiRepository.
- updates the pools subgraph service to pull using the SDK PoolsSubgraphRepository. 
- adds the new GraphQLQueryBuilder and switches all graph queries to it so that they can be fetched via the subgraph or the balancer API. 
- Changes Pools + APR's to use the new types from the SDK.

This is currently blocked by: https://github.com/balancer-labs/frontend-v2/pull/2335. Once that PR is merged this one can be updated and merged. 

It also requires the following SDK and API PR's to be merged:
- https://github.com/balancer-labs/balancer-sdk/pull/226
- https://github.com/balancer-labs/balancer-api/pull/130
- https://github.com/balancer-labs/balancer-api/pull/131

## Type of change


## How should this be tested?

- Ensure the initial pools page loads correctly
- Ensure individual pool pages load correctly
- Ensure all pool values are the same as they currently are in production. 
- Test if pool creation still works as expected
- Test if migrate flows still work as expected

## Checklist:
